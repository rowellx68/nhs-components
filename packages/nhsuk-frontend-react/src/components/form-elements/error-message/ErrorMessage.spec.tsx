import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, { Default as DefaultStory } from './ErrorMessage.stories';

const Default = composeStory(DefaultStory, meta);

it('should render the error message component', () => {
  const { container } = render(<Default />);

  expect(container.querySelector('.nhsuk-u-visually-hidden')).toHaveTextContent(
    'Error:',
  );
  expect(container).toMatchSnapshot();
});
