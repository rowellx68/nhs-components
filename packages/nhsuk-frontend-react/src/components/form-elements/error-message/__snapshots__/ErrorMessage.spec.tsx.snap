// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the error message component 1`] = `
<div>
  <span
    class="nhsuk-error-message"
  >
    <span
      class="nhsuk-u-visually-hidden"
    >
      Error: 
    </span>
    Date of birth must be in the pasts
  </span>
</div>
`;
